content/en/development/design-docs/clb-implementation.md

```diff
@@ -0,0 +1,62 @@
+---
+title: "StandardV2 Load Balancer - Container Based Backendpool"
+linkTitle: "StandardV2 Load Balancer - Container Based Backendpool"
+type: docs
+description: >
+    Implement new Load Balancer sku - StandardV2
+---
+
+# Container Load Balancer Design Document
+
+## Background
+
+> The following `User` stands for human users, or any cluster provisioning tools (e.g., AKS).
+
+This design document describes the user-facing design and workflow of the Standard V2 LoadBalancer - Container Based Backendpool.
+
+## Comparison to Node Based Load Balancer
+
+- In its current form, the **Software Load Balancer (SLB)** only supports load balancing to nodes. In a container cluster environment, there are multiple processing units or pods on a single node. To ensure that a user's request reaches an application within a pod, the request is first sent to the SLB, which determines the most suitable node. At the node level, the traffic is further balanced among the multiple pods. This process may involve multiple hops between pods until the correct node is found, especially when externalTrafficPolicy=Cluster is set. Consequently, the traffic is load balanced multiple times before reaching the desired pod.
+
+- In contrast, the **Container Load Balancer (CLB)** offers a fundamentally improved approach. The goal is to create an IP subnet overlay for all the pods that are part of a specific application. This means that when a user sends a request to the SLB, the SLB can directly route the traffic to the desired pod. This approach significantly reduces latency and provides a platform for new features, such as adding rule-based identities. In this context, the middleware capabilities of the cloud provider are utilized to construct and synchronize the Kubernetes cluster structure and its corresponding subnet overlay.
```

Comment:
Does container lb support custom cni like cillium? If we can post routing info via bgp?